src/UnoFlipController.java

```diff
@@ -86,42 +105,62 @@ public void actionPerformed(ActionEvent e) {
                         null,
                         null);
 
-                Card.Colour colour;
-                if (result == JOptionPane.OK_OPTION) {
+                // Colour is selected and OK is pressed
+                if (result == JOptionPane.OK_OPTION)
+                {
+                    // Colour to set the new current colour of the game
+                    Card.Colour colour;
+                    // Parse the input given by the user from the drop-down to set the new colour
                     String option = (String) comboBox.getSelectedItem();
                     colour = Card.Colour.valueOf(option);
+                    // Sets the new current colour
                     this.model.setCurrentColour(colour);
+                    // Notify view to show status of the current colour
                     this.model.notifyViews();
                     break;
-                } else if (result == JOptionPane.CLOSED_OPTION){
-                    continue;
                 }
             }
 
-        } else if(e.getActionCommand().equals(UnoFlipViewFrame.DRAW_CMD)){
+        }
+        // Draw card command for the player to draw a card
+        else if(e.getActionCommand().equals(UnoFlipViewFrame.DRAW_CMD))
+        {
+            // Model playTurn is called with -1 to signify a self draw one
             this.model.playTurn(-1);
         }
-        else if(e.getActionCommand().equals(UnoFlipViewFrame.NEXT_CMD)) {
+        // Next turn command to go to the next player
```

Comment:
Same idea: what actually happened was "A player hit the "next turn" button", and it can be placed inside the conditional block